app/observers/project_observer.rb

```diff
@@ -96,7 +96,13 @@ def notify_users(project)
 
   def request_refund_for_failed_project(project)
     project.payments.with_state('paid').each do |payment|
-      payment.request_refund
+      if payment.slip_payment?
```

Comment:
this logic should be in Payment model.
we can code it as a condition in the request_refund transition:
```ruby
class Payment < ActiveRecord::Base
  ...
  state_machine :state, initial: :pending do
    ...
    event :request_refund do
      transition paid: :pending_refund, if: ->(payment) {
        payment.can_request_refund?
      }

    end
    ...
  end
  ...
  private
  def can_request_refund?
    !payment.slip_payment? || payment.user.try(:bank_account).try(:valid?)
  end
end
```